aptos-move/aptos-vm-types/src/output.rs

```diff
@@ -126,24 +126,28 @@ impl VMOutput {
         delta_writes: Vec<(StateKey, WriteOp)>,
     ) -> TransactionOutput {
         let (change_set, gas_used, status) = self.unpack();
-        let (write_set, mut delta_change_set, events) = change_set.unpack();
-        let mut write_set_mut = write_set.into_mut();
 
         // We should have a materialized delta for every delta in the output.
-        assert_eq!(delta_writes.len(), delta_change_set.len());
+        assert_eq!(delta_writes.len(), change_set.delta_change_set().len());
+        debug_assert!(
+            delta_writes
+                .iter()
+                .all(|(k, _)| change_set.delta_change_set().contains(k)),
+            "Delta writes contain a key which does not exist in DeltaChangeSet."
+        );
 
         // Add the delta writes to the write set of the transaction.
-        delta_writes.into_iter().for_each(|item| {
-            debug_assert!(
-                delta_change_set.remove(&item.0).is_some(),
-                "Delta writes contain a key which does not exist in DeltaChangeSet."
-            );
-            write_set_mut.insert(item)
-        });
-
+        let mut write_set_mut = WriteSetMut::new(delta_writes);
```

Comment:
this is the part that will become much more complicated in v2, i.e. go through deserialization/serialization path to patch values?